sstables/sstable_set.cc

```diff
@@ -27,11 +27,29 @@
 
 namespace sstables {
 
-void sstable_run::insert(shared_sstable sst) {
+sstable_run::interval_type sstable_run::make_interval(const shared_sstable& sst) {
+    auto s = sst->get_schema();
+    return interval_type::closed(
+        compatible_ring_position_or_view(s, dht::ring_position(sst->get_first_decorated_key())),
+        compatible_ring_position_or_view(s, dht::ring_position(sst->get_last_decorated_key())));
+}
+
+bool sstable_run::insert(shared_sstable sst) {
+    auto interval = make_interval(sst);
+    auto r = _interval_set.equal_range(interval);
+    // overlapping detected, so sstable cannot satisfy the invariant.
+    if (r.first != r.second) {
```

Comment:
As someone who is not familiar with boost interval_set, I don't understand what this condition does - what is an "equal_range" and why does it detect overlap?
The comment helps, though, so I'll just believe it :-)

Reply:
equal_range is a standard C++ algorithm, and boost's interval_set::equal_range is just another implementation.